earth_enterprise/src/fusion/autoingest/StorageManager.h

```diff
@@ -338,6 +338,10 @@ bool StorageManager<AssetType>::SaveDirtyToDotNew(
   std::lock_guard<std::recursive_mutex> lock(storageMutex);
   typename std::map<AssetKey, PointerType>::iterator entry = dirtyMap.begin();
   while (entry != dirtyMap.end()) {
+    if ((entry->second.use_count() > 2) && writeDirty)  {
```

Comment:
I think there's a small bug here - if we're calling SaveDirtyToDotNew from ApplyPending, we want to always write out everything in the dirty map, even if `writeDirty` is true. That said, I don't think it affected your testing in any way.

Reply:
I don't believe this condition is met when SaveDirtyToDotNew is called in ApplyPending. Running the same build resulted in no assets having a use count greater than 2 when called from ApplyPending before this condition was introduced.